### DISCUSSION TITLE
Separation of specials in renpycompat

### DISCUSSION THREAD

**Discussion by madeddy** [CensoredUsername/unrpyc#191] [General]
First, i did not touch _un.rpyc_ for this. Editing _compile-unrpyc.py_ would be no prob but _compile.py_ is a question mark where to edit and i think i would perhaps break it.

While at this something came up: The code for `class PyExpr` in **un.rpyc** differs from the other as the `__getnewargs__` method is in the injector missing. Must this be or is this method overall unneeded?

P.s.: The first commit couldn't be smaller as everything interconnected is.

**Comment by CensoredUsername** [comment]
I'll handle un.rpyc, as the whole "create a python package in software" is a bit finicky. 

>While at this something came up: The code for class PyExpr in un.rpyc differs from the other as the __getnewargs__ method is in the injector missing. Must this be or is this method overall unneeded?

`__getnewargs__` is used during pickling. It returns the arguments that need to be passed to __new__ on this class to construct an object during unpickling. Pickling only happens in the translate module in unrpyc. un.rpyc doesn't ship with that, so it had no real need for that function.

**Comment by madeddy** [comment]
> I'll handle un.rpyc, as the whole "create a python package in software" is a bit finicky.

Quite so. Thanks.
